### PR TITLE
Exclude files flag

### LINKED ISSUE #193: Can we please add a feature to only run mutation tests on certain files [request]
# 🚀 Feature Request

### Is your feature request related to a problem? Please describe
None

### Describe the solution you'd like
I'd really love a command line option such as --paths-to-mutate or --files-to-mutate where we only run on a subset of the files in a repo instead of running on every file. You pass in a comma separated list of files and those are the only files that are mutated.

### Describe alternatives you've considered
Using a different go mutations library

### Additional context
I'd like to run gremlins on the files changed in a pull request, so that gremlins can be used on larger repos.

Examples from other mutation libraries:  
https://github.com/muter-mutation-testing/muter
https://github.com/avito-tech/go-mutesting#how-do-i-use-go-mutesting

### PR DESCRIPTION
**Pull request by rusinikita**
## Proposed changes

Flag to exclude files from mutation tests.

It use full to exclude generated and abandoned code and test utils.

## Types of changes

New flag `exclude-files`

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/go-gremlins/gremlins/docs/CONTRIBUTING.md) doc
- [x] Lint and unit tests pass locally with my changes (`make all`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

I added []string values support in flags package and fixed few lived mutants.

Closes #193

### CODE REVIEW

**Comment by rusinikita** [comment]
@k3rn31 Hi! Please review few PRs?

In this PR actions not started, that is strange. But linter and tests runs [successful](https://github.com/rusinikita/gremlins/actions/runs/7861648184/job/21450232449?pr=2)

**Comment by rusinikita** [comment]
@k3rn31 Sorry, I know you don't have much time. All checks green. Just tell when do you have time for review?

**Comment by k3rn31** [comment]
> @k3rn31 Sorry, I know you don't have much time. All checks green. Just tell when do you have time for review?

Hi, yes, I'm sorry I'm not very responsive these days.
I added a little review note a few days ago about the usage message of the new flag, can you fix it?

**Comment by rusinikita** [comment]
> I added a little review note a few days ago about the usage message of the new flag, can you fix it?

I do not see any. Have you pressed `finish review` button?

**Review by k3rn31 — CHANGES_REQUESTED** [review]

File: cmd/unleash.go [hunk 1]
```diff
@@ -203,6 +211,7 @@ func setFlagsOnCmd(cmd *cobra.Command) error {
 		{Name: paramDiff, CfgKey: configuration.UnleashDiffRef, Shorthand: "D", DefaultV: "", Usage: "diff branch or commit"},
 		{Name: paramOutput, CfgKey: configuration.UnleashOutputKey, Shorthand: "o", DefaultV: "", Usage: "set the output file for machine readable results"},
 		{Name: paramIntegrationMode, CfgKey: configuration.UnleashIntegrationMode, Shorthand: "i", DefaultV: false, Usage: "makes Gremlins run the complete test suite for each mutation"},
+		{Name: paramExcludeFiles, CfgKey: configuration.UnleashExcludeFiles, Shorthand: "E", DefaultV: []string{}, Usage: "bibibibibibibibibibibibibbibibibi"},
```
Comment (giose86) [inline cmd/unleash.go]: The usage statement needs a more detailed explanation :)

Comment (rusinikita) [reply cmd/unleash.go]: Yes) Fixed it

**Review by k3rn31 — APPROVED** [review]

**Pull request merged**